api/src/main/resources/marquez/db/migration/V48__dataset_symlinks.sql

```diff
@@ -2,7 +2,7 @@
 
 CREATE TABLE dataset_symlinks (
   dataset_uuid      UUID,
-  name              VARCHAR(255) NOT NULL,
+  name              VARCHAR NOT NULL,
   namespace_uuid    UUID REFERENCES namespaces(uuid),
   type              VARCHAR(64),
   is_primary        BOOLEAN DEFAULT FALSE,
```

Comment:
The insert statement in V48__dataset_symlinks.sql line#16 would have inserted partial data successfully so we also need to modify the insert to fail gracefully on duplicate keys and continue inserting missing data.

Reply:
I think the insert job would fail atomically and rollback, so users that face this problem shouldn't have any partial data. However, we should add an insert statement in `V52` and just add an `ON CONFLICT DO NOTHING` directive to ensure that everything in the `datasets` table has a value in the `dataset_symlinks`.

Reply:
My understanding of Flyway scripts (based on [docs](https://flywaydb.org/documentation/learnmore/faq#rollback):

> Flyway runs each migration in a separate transaction. In case of failure this transaction is rolled back. 

 * If a migration fails, nothing has been changed in DB, you just need to run migration script again with a fixed code.
 * If a migration succeeded (which means one did not have long names), we add extra migration to extend field length and don't need to insert anything.

Reply:
Ok, let's leave it as is and trust Flyway to behave as expected. I've seen unexpected behaviors in production deployments, so I don't entirely trust our setup, but I can't pinpoint a specific issue. This should work for most users, and I don't want to unnecessarily burden all users with a potentially costly INSERT migration without a specific issue to point to.